src/sql/workbench/contrib/query/browser/queryActions.ts

```diff
@@ -594,6 +594,10 @@ export class ListDatabasesActionItem extends Disposable implements IActionViewIt
 			placeholder: this._selectDatabaseString,
 			ariaLabel: this._selectDatabaseString
 		});
+
+		// Allows database selector to commit typed or pasted DB names without the need to click
+		// or press enter to make a selection when focus is moved away from the selector.
+		this._register(this._dropdown.onBlur(() => this.databaseSelected(this._dropdown.value)));
```

Comment:
call to the this.databaseSelected method will trigger a changeDatabase operation which is expensive, we should modify it to only do it when the database is actually changed, especially when the user is only navigating using keyboard.

Reply:
I will go back and make the proper updates to this.

Reply:
I moved the requested check to the databaseSelected method, so that we're only committing the change when the DB name has changed.